api.go

```diff
@@ -51,21 +51,21 @@ func CreatePod(podConfig PodConfig) (*Pod, error) {
 	}
 
 	// Initialize the network.
-	err = p.network.init(&(p.config.NetworkConfig))
+	netNsPath, netNsCreated, err := p.network.init(p.config.NetworkConfig)
 	if err != nil {
 		return nil, err
 	}
 
 	// Execute prestart hooks inside netns
-	err = p.network.run(p.config.NetworkConfig.NetNSPath, func() error {
+	err = p.network.run(netNsPath, func() error {
```

Comment:
Given the docker pid vs tid and netns issue, I wonder if we should execute the hook using reexec. I am not sure if docker tracks the pid for later use (I think not). That will eliminate the issue we have with docker. It may also simplify our code longer term.